MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionPresenter.cpp

```diff
@@ -1325,5 +1351,209 @@ void EnggDiffractionPresenter::calcVanadiumWorkspaces(
   vanCurvesWS = ADS.retrieveWS<MatrixWorkspace>(curvesName);
 }
 
+/**
+ * Checks the plot type selected and applies the appropriate
+ * python function to apply during first bank and second bank
+ *
+ * @param outWSName title of the focused workspace
+ * @param bank the number of bank
+ */
+void EnggDiffractionPresenter::plotFocusedWorkspace(std::string outWSName,
+                                                    std::string bank) {
+  const bool plotFocusedWS = m_view->focusedOutWorkspace();
+  int plotType = m_view->currentPlotType();
+  if (plotFocusedWS) {
+    if (plotType == 0) {
+      if (bank == "1")
+        m_view->plotFocusedSpectrum(outWSName);
+      if (bank == "2")
+        m_view->plotReplacingWindow(outWSName);
+    } else if (1 == plotType) {
+      if (bank == "1")
+        m_view->plotFocusedSpectrum(outWSName);
+      if (bank == "2")
+        m_view->plotWaterfallSpectrum(outWSName);
+    } else if (2 == plotType) {
+      m_view->plotFocusedSpectrum(outWSName);
+    }
+  }
+}
+
+/**
+ * Convert the generated output files and saves them in
+ * FocusedXYE format
+ *
+ * @param inputWorkspace title of the focused workspace
+ * @param bank the number of the bank as a string
+ * @param runNo the run number as a string
+ */
+void EnggDiffractionPresenter::saveFocusedXYE(const std::string inputWorkspace,
+                                              std::string bank,
+                                              std::string runNo) {
+  auto alg = Algorithm::fromString("SaveFocusedXYE");
+  std::string fullFilename;
+  if (inputWorkspace.std::string::find("texture") != std::string::npos) {
+    fullFilename = "ENGINX_" + runNo + "_texture_" + bank + ".dat";
+  }
+  if (inputWorkspace.std::string::find("cropped") != std::string::npos) {
+    fullFilename = "ENGINX_" + runNo + "_cropped_" +
+                   boost::lexical_cast<std::string>(g_croppedCounter) + ".dat";
+    g_croppedCounter++;
+  } else {
+    fullFilename = "ENGINX_" + runNo + "_bank_" + bank + ".dat";
+  }
+  std::string saveDir;
+  try {
+    // takes to the root of directory according to the platform
+    // and appends the following string provided
+    saveDir =
+        Poco::Path(saveDir).expand("/EnginX_Mantid/User/" + runNo + "/Focus/");
+    if (!Poco::File(saveDir).exists()) {
+      Poco::File(saveDir).createDirectories();
+    }
+  } catch (std::runtime_error &re) {
```

Comment:
With the poco path issues I got crashes with the exception `Poco::FileAccessDeniedException`, so together with this catch it would be a good idea to add one more specific catch (first):
```
catch (Poco::FileAccessDeniedException &e) {
// print more specific error message - there's a file access/permission error
}
```